src/main/java/redis/clients/jedis/JedisFactory.java

```diff
@@ -26,7 +26,11 @@ public JedisFactory(final String host, final int port, final int connectionTimeo
     this.hostAndPort.set(new HostAndPort(host, port));
     this.connectionTimeout = connectionTimeout;
     this.soTimeout = soTimeout;
-    this.password = password;
+    if(password != null && ! password.trim().isEmpty()){
```

Comment:
trim is not necessary as if the string has white spaces, `isEmpty` returns false.
Please fix braces indentation.